darwin/DarwinProcessList.c

```diff
@@ -160,9 +160,11 @@ void ProcessList_delete(ProcessList* this) {
    free(this);
 }
 
-static double ticksToNanoseconds(const double ticks) {
+// Converts "scheduler ticks" to nanoseconds.
+// See `sysconf(_SC_CLK_TCK)`, as used to define the `Platform_clockTicksPerSec` constant.
+static double schedulerTicksToNanoseconds(const double ticks) {
    const double nanos_per_sec = 1e9;
-   return (ticks / Platform_timebaseToNS) * (nanos_per_sec / (double) Platform_clockTicksPerSec);
+   return ticks * (nanos_per_sec / (double) Platform_clockTicksPerSec); 
```

Comment:
The values that come back from `host_processor_info` are *NOT* in Mach timebase tick units.

They're in "scheduler ticks" (as I call them), which are still the same 100 ticks/second on M1 as they used to be on Intel Macs.

Converting with `Platform_timebaseToNS` here was mistake, but it cancels out after division, ended up making the percentage correct, but not the "CPU time" stat.

Reply:
Given that this conversion function is the only use of `Platform_clockTicksPerSec`, I suggest we refactor:

1. Hide the `Platform_clockTicksPerSec` constant from `Platform.h`
2. Move `schedulerTicksToNanoseconds` to `Platform.c`
3. Expose `schedulerTicksToNanoseconds` from `Platform.h`, and call that from here.

Reply:
P.s. accidental trailing space here. I'll remove it with my "touch up" commit.

Reply:
> The values that come back from `host_processor_info` are _NOT_ in Mach timebase tick units.
> 
> They're in "scheduler ticks" (as I call them), which are still the same 100 ticks/second on M1 as they used to be on Intel Macs.

:fire: :fire: :fire:

Reply:
> P.s. accidental trailing space here. I'll remove it with my "touch up" commit.

Just amend/squash them in …

Reply:
> Given that this conversion function is the only use of `Platform_clockTicksPerSec`, I suggest we refactor:
> 
>     1. Hide the `Platform_clockTicksPerSec` constant from `Platform.h`
> 
>     2. Move `schedulerTicksToNanoseconds` to `Platform.c`
> 
>     3. Expose `schedulerTicksToNanoseconds` from `Platform.h`, and call that from here.

Done: https://github.com/htop-dev/htop/pull/766